test/extended/node/oom_detection.go

```diff
@@ -0,0 +1,45 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO: expand this test to SSH into every node in the cluster and gather kubelet logs in case kubelet doesn't send
+// 		 events to the api-server or `oc adm node-logs`
+var _ = g.Describe("[sig-node][Late]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("no-oom-kills")
+
+	g.It("should not have pods that are OOM Killed", func() {
+		kubeClient := oc.AdminKubeClient()
+
+		events, err := kubeClient.CoreV1().Events("").List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			g.Fail(fmt.Sprintf("Unexpected error: %v", err))
+		}
+
+		oomEncounteredEvents := []string{}
+		errorMsg := "System OOM encountered"
+		for _, event := range events.Items {
+			if !strings.Contains(event.Message, errorMsg) {
+				continue
+			}
+			oomEncounteredEvent := fmt.Sprintf("OOM encountered within node %v", event.InvolvedObject.Name)
```

Comment:
Do the events say what got OOMed?  That would make it easier to route to the team in charge of the greedy component (for cases where OOM picked the right greedy component).

Reply:
I thought about it but the message only talks about the [victim process](https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/pkg/kubelet/oom/oom_watcher_linux.go#L79). Not sure, if it be beneficial

Reply:
The process that ends up being the cause of the  OOM Killer invocation isn't always at fault. Any memory allocation by any process could cause the oom killer to be invoked.